src/main/webapp/app/tutor-exercise-dashboard/tutor-exercise-dashboard.component.html

```diff
@@ -310,13 +309,24 @@ <h2>
                                     <button
                                         *ngIf="complaint.accepted === undefined; else continueButton"
                                         class="btn btn-success btn-sm"
-                                        (click)="openAssessmentEditor(complaint.result.submission.id, false)"
+                                        (click)="
+                                            exercise.type === ExerciseType_PROGRAMMING
+                                                ? openManualResultDialog(complaint.result.participation.id, complaint.result)
+                                                : openAssessmentEditor(complaint.result.submission.id, false)
+                                        "
                                     >
                                         <fa-icon icon="folder-open" [fixedWidth]="true"></fa-icon>
                                         {{ 'artemisApp.tutorExerciseDashboard.evaluateComplaint' | translate }}
                                     </button>
                                     <ng-template #continueButton>
-                                        <button class="btn btn-primary btn-sm" (click)="openAssessmentEditor(complaint.result.submission.id, false)">
+                                        <button
+                                            class="btn btn-primary btn-sm"
+                                            (click)="
+                                                exercise.type === ExerciseType_PROGRAMMING
+                                                    ? openManualResultDialog(complaint.result.participation.id, complaint.result)
+                                                    : openAssessmentEditor(complaint.result.submission.id, false)
```

Comment:
I am wondering if we can move this as one method into TypeScript and reuse this four times in the HTML.

(Also, see comment in `tutor-exercise-dashboard.component.ts` on change in parameters for `openManualResultDialog`.)

Reply:
The idea was to not parameters passed into the `openAssessmentEditor` since it is used like 7 times. But I can also do that.

Reply:
I am wondering if we should add a method like this:

```typescript
handleComplaint(complaint: Complaint) {
  if (this.exercise.type === ExerciseType.PROGRAMMING) {
    this.openManualResultDialog(complaint.result);
  } else {
    this.openAssessmentEditor(complaint.result.submission!.id, false)
  }
}
```

and thereby reduce the HTML to this:

```html
<button
  class="btn btn-primary btn-sm"
  (click)="handleComplaint(complaint)"
>
  ...
</button>
```
OR
```html
<button
  class="btn btn-primary btn-sm"
  (click)="handleComplaint(moreFeedbackRequest)"
>
  ...
</button>
```

Reply:
Great idea! Added it and called it `viewComplaint`